examples/src/examples/misc/gizmo.example.mjs

```diff
@@ -0,0 +1,156 @@
+import * as pc from 'playcanvas';
+import { deviceType, rootPath } from 'examples/utils';
+
+const canvas = /** @type {HTMLCanvasElement} */ (document.getElementById('application-canvas'));
+window.focus();
+
+const gfxOptions = {
+    deviceTypes: [deviceType],
+    glslangUrl: rootPath + '/static/lib/glslang/glslang.js',
+    twgslUrl: rootPath + '/static/lib/twgsl/twgsl.js'
+};
+
+const device = await pc.createGraphicsDevice(canvas, gfxOptions);
+device.maxPixelRatio = Math.min(window.devicePixelRatio, 2);
+
+const createOptions = new pc.AppOptions();
+createOptions.graphicsDevice = device;
+createOptions.mouse = new pc.Mouse(document.body);
+createOptions.keyboard = new pc.Keyboard(window);
+
+createOptions.componentSystems = [
+    pc.RenderComponentSystem,
+    pc.CameraComponentSystem,
+    pc.LightComponentSystem,
+    pc.ScriptComponentSystem
+];
+createOptions.resourceHandlers = [pc.TextureHandler, pc.ContainerHandler, pc.ScriptHandler, pc.FontHandler];
+
+const app = new pc.AppBase(canvas);
+app.init(createOptions);
+
+// set the canvas to fill the window and automatically change resolution to be the same as the canvas size
+app.setCanvasFillMode(pc.FILLMODE_FILL_WINDOW);
+app.setCanvasResolution(pc.RESOLUTION_AUTO);
+
+// load assets
+const assets = {
+    script: new pc.Asset('script', 'script', { url: rootPath + '/static/scripts/camera/orbit-camera.js' }),
+    font: new pc.Asset('font', 'font', { url: rootPath + '/static/assets/fonts/courier.json' })
+};
+/**
+ * @param {pc.Asset[] | number[]} assetList - The asset list.
+ * @param {pc.AssetRegistry} assetRegistry - The asset registry.
+ * @returns {Promise<void>} The promise.
+ */
+function loadAssets(assetList, assetRegistry) {
+    return new Promise((resolve) => {
+        const assetListLoader = new pc.AssetListLoader(assetList, assetRegistry);
+        assetListLoader.load(resolve);
+    });
+}
+await loadAssets(Object.values(assets), app.assets);
+
+app.start();
+/**
+ * @param {pc.Color} color - The color.
+ * @returns {pc.Material} - The standard material.
+ */
+function createColorMaterial(color) {
+    const material = new pc.StandardMaterial();
+    material.diffuse = color;
+    material.update();
+    return material;
+}
+
+// scene settings
+app.scene.ambientLight = new pc.Color(0.2, 0.2, 0.2);
+
+// create entities
+const box = new pc.Entity('box');
+box.addComponent('render', {
+    type: 'box',
+    material: createColorMaterial(new pc.Color(0.8, 0.8, 0.8))
+});
+app.root.addChild(box);
+
+// create camera entity
+const camera = new pc.Entity('camera');
+camera.addComponent('camera', {
+    clearColor: new pc.Color(0.1, 0.1, 0.1),
+    farClip: 1000
+});
+camera.addComponent('script');
+const orbitCamera = camera.script.create('orbitCamera');
+camera.script.create('orbitCameraInputMouse');
+camera.script.create('orbitCameraInputTouch');
+camera.setPosition(1, 1, 1);
+app.root.addChild(camera);
+orbitCamera.distance = 5 * camera.camera?.aspectRatio;
```

Comment:
Isn't `camera.camera` guaranteed to be valid at this point?

Reply:
It is but typescript doesnt know that